.dockerignore

```
@@ -0,0 +1,7 @@
+.git
+__pycache__/
+*.py[cod]
+input
+models
+notebooks
+output
```

An alternative suggestion, ignore everything and only specify patterns for what to include?

```suggestion
*
!app/
!comfy/
!comfy_extras/
!custom_nodes/
!web/
!requirements.txt
!*.py
!config.toml
```

Otherwise it'd probably be similar to `.gitignore`? Not too fussed about the contents here, just that `.gitignore` is to filter out unwanted stuff from being committed to the repo, whereas with a fresh `git clone` the `.dockerignore` is to filter out what you don't want copied over to the image context (_for later use with `COPY`_).

The suggestion I've provided is a bit more clearer at what you're actually interested in keeping for bringing into the image. Presently other content is brought in otherwise, but perhaps `tests` + `tests-ui` is still wanted in the image? (_or files like `extra_model_paths.yaml.example`?_)

---

> The .git folders for ComfyUI and all custom nodes must be preserved.
> Custom nodes or ComfyUI without a .git folder are no longer under the Manager's management.

This is what the maintainer of  [ComfyUI-Manager](https://github.com/ltdrdata/ComfyUI-Manager) said. So I guess it'll better to keep `.git` folder

---

Any git updates won't survive the container restart. I think it might be misleading, to be able to update ComfyUI via the manager, and then restart and be at the old version again.

The only benefit of having the git index in the image would be that ComfyUI-Manager could tell the ComfyUI version.

---

> The only benefit of having the git index in the image would be that ComfyUI-Manager could tell the ComfyUI version.

It is already a think. Without it, the manager will tell only "ComfyUI outdated".

I think that every docker user knows that containers do not persist data if not instructed otherwise thus update should be handled in other ways (e.g. docker tags)

---

Personally I'm not a fan of software managing its core itself, (for my mailcow setup I simply run the official update.sh and that does all the pulling and downs and ups for me) but here's an idea:

Include the git repo but also give the container a persisted volume. When starting the container the .git folder is copied over to the volume and further management shenanigans can be done from there. Obviously needs an EVN flag to turn this on.